subsys/bluetooth/mesh/shell/shell_loc_cli.c

```diff
@@ -0,0 +1,190 @@
+
+#include "mesh/net.h"
+#include "mesh/access.h"
+#include "shell_utils.h"
+
+static struct bt_mesh_model *mod;
+
+static int shell_loc_global_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_GEN_LOCATION_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_loc_cli *cli = mod->user_data;
+	struct bt_mesh_loc_global rsp;
+
+	int err = bt_mesh_loc_cli_global_get(cli, NULL, &rsp);
+
+	if (!err) {
+		shell_print(shell, "Latitude: %f, longitude: %f, altitude: %d", rsp.latitude,
+			    rsp.longitude, rsp.altitude);
+	}
+
+	return err;
+}
+
+static int global_set(const struct shell *shell, size_t argc, char *argv[], bool acked)
+{
+	double latitude = shell_model_str2dbl(shell, argv[1]);
+	double longitude = shell_model_str2dbl(shell, argv[2]);
+	int16_t altitude = (int16_t)strtol(argv[3], NULL, 0);
+
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_GEN_LOCATION_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_loc_cli *cli = mod->user_data;
+	struct bt_mesh_loc_global set = {
+		.latitude = latitude,
+		.longitude = longitude,
+		.altitude = altitude,
+	};
+
+	if (acked) {
+		struct bt_mesh_loc_global rsp;
+		int err = bt_mesh_loc_cli_global_set(cli, NULL, &set, &rsp);
+
+		if (!err) {
+			shell_print(shell, "Latitude: %f, longitude: %f, altitude: %d",
+				    rsp.latitude, rsp.longitude, rsp.altitude);
+		}
+
+		return err;
+	} else {
+		return bt_mesh_loc_cli_global_set_unack(cli, NULL, &set);
+	}
+}
+
+static int shell_loc_global_set(const struct shell *shell, size_t argc, char *argv[])
+{
+	return global_set(shell, argc, argv, true);
+}
+
+static int shell_loc_global_set_unack(const struct shell *shell, size_t argc, char *argv[])
+{
+	return global_set(shell, argc, argv, false);
+}
+
+static int shell_loc_local_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_GEN_LOCATION_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_loc_cli *cli = mod->user_data;
+	struct bt_mesh_loc_local rsp;
+
+	int err = bt_mesh_loc_cli_local_get(cli, NULL, &rsp);
+
+	if (!err) {
+		shell_print(shell,
+			    "North: %d, east: %d, altitude: %d, "
+			    "floor_number: %d, is_mobile: %d, "
+			    "time_delta: %d, precision_mm: %d",
+			    rsp.north, rsp.east, rsp.altitude, rsp.floor_number, rsp.is_mobile,
+			    rsp.time_delta, rsp.precision_mm);
+	}
+
+	return err;
+}
+
+static int local_set(const struct shell *shell, size_t argc, char *argv[], bool acked)
+{
+	int16_t north = (int16_t)strtol(argv[1], NULL, 0);
+	int16_t east = (int16_t)strtol(argv[2], NULL, 0);
+	int16_t altitude = (int16_t)strtol(argv[3], NULL, 0);
+	int16_t floor = (int16_t)strtol(argv[4], NULL, 0);
+	int32_t time_delta = (argc >= 6) ? (int32_t)strtol(argv[5], NULL, 0) : 0;
+	uint32_t precision_mm = (argc >= 7) ? (uint32_t)strtol(argv[6], NULL, 0) : 0;
+	bool is_mobile = (argc == 8) ? (bool)strtol(argv[7], NULL, 0) : false;
```

Comment:
```suggestion
	bool is_mobile = (argc >= 8) ? (bool)strtol(argv[7], NULL, 0) : false;
```
There are more of them. So we should discuss if we do this or not.

Reply:
Yes, we should agree before I do it. There is nothing wrong in changing it, and in a way it removes a potential bug src for this module by introducing a generic way of including optional args. However I got this gut feeling that it is wrong to allow the variable to accept a `argc` value that will be prohibited by the shell module, so im not certain what is the best solution here.